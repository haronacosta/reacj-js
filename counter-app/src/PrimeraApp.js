import React from 'react';

const PrimeraApp = ({ saludo = 'Hola soy el valor por defecto' }) => {
  return (
    <>
      <h1>{saludo}</h1>
      <p>Parrafo de mi primera app</p>
    </>
  );
};

export default PrimeraApp;
