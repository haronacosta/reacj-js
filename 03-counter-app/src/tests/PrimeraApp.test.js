import React from 'react';
import { shallow } from 'enzyme';
import PrimeraApp from '../PrimeraApp';

describe('Pruebas en "<PrimeraApp />"', () => {
  // test('Debe de regresarme Hola, soy Goku', () => {
  //   const saludo = `Hola, soy Goku`;
  //   const { getByText } = render(<PrimeraApp saludo={saludo} />);
  //   expect(getByText(saludo)).toBeInTheDocument();
  // });

  test('Se debe mostrar correctamente <PrimeraApp />', () => {

    const saludo = 'Hola, soy Goku';

    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('debe mostrar el subtitulo enviado por props', () => {

    const saludo = 'Hola, soy Goku';

    const subtitulo = 'Soy un subtitulo';

    // Shallow es la manera de manejar los componentes en enzyme

    const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo} />);

    // .find() es el equivalente al querySelector de JS pero con varios metodos
    const textoSubtitulo = wrapper.find('p').text();

    console.log(textoSubtitulo);

    expect(textoSubtitulo).toBe(subtitulo);

  });
});
