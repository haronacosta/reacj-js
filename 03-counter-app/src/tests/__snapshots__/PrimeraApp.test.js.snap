// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en "<PrimeraApp />" Se debe mostrar correctamente <PrimeraApp /> 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
  </h1>
  <p />
</Fragment>
`;
